pkg/plugin/manifest/convert.go

```diff
@@ -129,6 +133,52 @@ func (i Extension) extension(pluginID id.PluginID, sys bool) (*plugin.Extension,
 	return ext, schema, nil
 }
 
+func (l *WidgetLayout) layout() *scene.WidgetLayout {
+	if l == nil {
+		return &scene.WidgetLayout{
+			Extended:   false,
+			Extendable: true,
+			Floating:   false,
+			DefaultLocation: &scene.WidgetLocation{
+				Zone:    "outer",
+				Section: "left",
+				Area:    "top",
+			},
+		}
+	}
+
+	var swl scene.WidgetLayout
+	swl.Extended = false
+	if l.Extendable == nil {
+		swl.Extendable = true
+	} else {
+		e := l.Extendable
+		swl.Extendable = *e
+	}
+	if l.Floating == nil {
+		swl.Floating = false
+	} else {
+		f := l.Floating
+		swl.Floating = *f
+	}
+	var dl scene.WidgetLocation
+	if l.DefaultLocation == nil {
+		dl = scene.WidgetLocation{
+			Zone:    "outer",
+			Section: "left",
+			Area:    "top",
+		}
+	} else {
+		dl = scene.WidgetLocation{
+			Zone:    l.DefaultLocation.Zone,
+			Section: l.DefaultLocation.Section,
+			Area:    l.DefaultLocation.Area,
+		}
+	}
+	swl.DefaultLocation = &dl
```

Comment:
```suggestion
	if dl == nil {
		swl.DefaultLocation = dl
	}
```

Empty struct always will be set

Reply:
Sorry, but why is this desirable? Currently, if the Widget doesn't have a DefaultLocation it will be set with `outer, left, top` as a system default. If it has a default location, it is set with that.
So `dl` should never be nil when being set to to `swl.DefaultLocation`.

Reply:
Hm, I confused because type of DefaultLocation field of WidgetLayout is a reference (`*DefaultLocation`) so it can be nil. If you do not want DefaultLocation to be nil, you can change the type to `DefaultLocation`, because WidgetLocation has no private fields so it can be treated as a simple data container (that is also a "value object").

Note: Strictly speaking, in DDD, the value object should be immutable, but we'll leave that aside for now

Reply:
Oh I see what you mean. Then perhaps it is better for all WidgetLayout fields to be mandatory, and just the WidgetLayout to be possibly nil (for non-widget plugins).
I'll update so that default location is necessary.

Reply:
I have given it more thought. And I think for plugin developers, it would be nice if all Widgetlayout fields are optional except float. Because if float is true, the widget doesn't need any of the other fields.
So I will update WidgetLayout in that way.